### PR TITLE
chore: DLT-1513 remove multiple exports

### PR DESCRIPTION
**Pull request by juliodialpad**
# Remove multiple exports

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will not increment the version number, but will still deploy to documentation site on release:

- [x] Build system

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1513?atlOrigin=eyJpIjoiZjY0NmMzYjE1Yzc4NDdhMGE5YmM4MTVlNmNiYWEwMzEiLCJwIjoiaiJ9

## :book: Description

- Removed exports from dialtone-vue2, dialtone-vue3 and dialtone
- Moved dialtone.css file from `lib/dist/css` to `lib/dist` to solve issues with fonts not being found on webpack + vue 2.6
- Removed `{ reactive }` import from DtHovercard as it is vue 2.7+ only and it will not work on vue 2.6
- Greatly reduced the bundle size by doing the following:
  - Created a new package dialtone-emojis to externalize this dependency
  - Externalized all dependencies

## :bulb: Context

We were starting to have a lot of exports on package.json which would be hard to maintain in the future, we needed to go back to having just two exports (main and style)

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [x] I have added / updated unit tests.
- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script.

If new component:

- I am exporting any new components or constants:
  - [x] from the index.js in the root (either `packages/dialtone-vue2` or `packages/dialtone-vue3`).

## :crystal_ball: Next Steps

- Add tree shake, this will be done on a separate PR due too many files changed already on this.

## :camera: Screenshots / GIFs
- Before
Screenshot 2024-03-20 at 12 07 52 a m
image

- After
Screenshot 2024-03-20 at 12 05 06 a m
Screenshot 2024-03-20 at 12 09 50 a m

## :link: Sources

- https://medium.com/@rjayakumarreddy887/create-a-ui-component-library-that-is-tree-shakable-cffbbc4d1829
- https://dev.to/rafaelogic/optimizing-vuejs-performance-a-guide-to-tree-shaking-with-webpack-and-vite-3if7
- https://blog.theodo.com/2021/04/library-tree-shaking/
- https://github.com/vitejs/vite/discussions/14676
- https://dev.to/receter/how-to-create-a-react-component-library-using-vites-library-mode-4lma

### CODE REVIEW

**Comment by iropolo** [comment]
Seems like a huge improvement, do this impact how to use the library that we should notify or all this is just internal refactor?

**Comment by juliodialpad** [comment]
> Seems like a huge improvement, do this impact how to use the library that we should notify or all this is just internal refactor?

I removed some exports: dialtone-vue/emoji, dialtone-vue/message_input and dialtone-vue/directives. Updated docs accordingly but a migration is going to be required.

Some of those exports might come back with the tree-shaking improvement I'm working on currently, I think I'll create a branch off of this one and finish the tree shaking improvement before we can merge this one.

**Comment by braddialpad** [comment]
> Some of those exports might come back with the tree-shaking improvement I'm working on currently, I think I'll create a branch off of this one and finish the tree shaking improvement before we can merge this one.

Yeah I agree we should do this.

File: packages/dialtone-vue2/vite.config.js [hunk 1]
```diff
@@ -10,21 +10,53 @@ const __dirname = path.dirname(__filename);
...
       external: [
+        /@dialpad/,
+        /@linusborg/,
+        /@tiptap/,
+        /date-fns/,
+        /emoji-regex/,
+        /emoji-toolkit/,
+        /tippy\.js/,
+        /prosemirror/,
         'vue',
```
Comment (braddialpad) [inline packages/dialtone-vue2/vite.config.js:53]: If we externalize all these dependencies doesn't that mean the consumer will have to manually install these in order to run the library?

Comment (juliodialpad) [reply packages/dialtone-vue2/vite.config.js:53]: Those are already listed under `dependencies` so npm/pnpm/yarn will install `dependencies` and `peerDependencies`

Comment (braddialpad) [reply packages/dialtone-vue2/vite.config.js:53]: Cool, didn't actually realize it worked like this. I guess that makes sense as vite is totally separate from npm so you have to tell it what to bundle

Comment (braddialpad) [reply packages/dialtone-vue2/vite.config.js:53]: Could also use a vite plugin so we don't have to update this manually: https://www.npmjs.com/package/vite-plugin-externalize-deps

Comment (juliodialpad) [reply packages/dialtone-vue2/vite.config.js:53]: Tried that plugin and didn't worked well, maybe I was doing something wrong, I'll give it another shot.

Also tried importing dependencies and peerDependencies from package.json but needed to use `assert { type: 'json' }` which is too experimental to rely on it.

Comment (braddialpad) [reply packages/dialtone-vue2/vite.config.js:53]: Yeah it's fine to do manually for now. not that big of deal

**Review by braddialpad — APPROVED** [review]
Looks good, wait on merging for other PR as mentioned.

**Review by iropolo — APPROVED** [review]

**Comment by juliodialpad** [comment]
Been having some issues with unstable visual tests due to lazy-loading icons, will investigate more about this later if they continue failing.

**Pull request merged**